src/kostentraeger/transformer.ts

```diff
@@ -0,0 +1,142 @@
+import { KTORInterchange, KTORMessage, ANS, ASP, DFU, KTO, UEM } from "./edifact/segments"
+import { 
+    Address, 
+    BankAccountDetails, 
+    Contact, 
+    Institution, 
+    InstitutionList, 
+    ReceiptTransmissionMethods
+} from "./types"
+
+export default function transform(interchange: KTORInterchange): InstitutionList {
+    return {
+        spitzenverbandIK: interchange.spitzenverbandIK,
+        creationDate: interchange.creationDate,
+        institutions: interchange.institutions.map((msg) => {
+            requirePreconditions(msg)
+            return transformMessage(msg)
+        }).filter((msg): msg is Institution => !!msg)
+    }
+}
+
+function requirePreconditions(msg: KTORMessage) {
+    /* Simplification: Certain data is documented that it could be different, but de-facto it's not.
+       Let's assert that it is never a value so that our data model and application logic can be 
+       simpler
+     */
+
+    const messageTxt = `Message ${msg.id} -`
+
+    if (msg.idk.institutionsart != "99") {
+        throw new Error(`${messageTxt} Expected that "institutionsart" is always 99`)
+    }
+    if (msg.idk.vertragskassennummer) {
+        throw new Error(`${messageTxt} Expected that "vertragskassennummer" is never set`)
+    }
```

Comment:
If this information is irrelevant to us anyway, can't we ignore it instead of throwing an exception?
I just imagine that we can't process a future Kostenträger file because the parser throws an error as soon as someone has set a value for Vertragskasssennummer. Is it feasible to be less strict in this case?

Reply:
Turns out that the vertragskassennummer is set for some after all anyway.

At this point, we (I) don't know yet if we really don't need that information. I was just cutting down on the data model to leave out things that are never set anyway. I throw exceptions so that in case it is set after all (like here), we are notified of a possible missing datum in our data model.

Anyway, will work on this next (parser vs real data)